CONTRIBUTING.md

```diff
@@ -62,6 +62,30 @@ In general, we try to limit adding third-party production dependencies. If one i
 
 We use eslint to enforce certain coding standards. Please see our [.eslintrc](./.eslintrc.js) file for specific rule configuration.
 
+### Commit Guidelines
+
+We use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) to format commit messages for this repository. Conventional Commits provide a standardized format for commit messages that allows for automatic generation of changelogs and easy tracking of changes.
+
+When contributing to this repository, please ensure that your commit messages adhere to the Conventional Commit guidelines. Specifically, your commit messages should:
+
+* Start with a type, indicating the kind of change being made (e.g. `feat` for a new feature, `fix` for a bugfix, etc.). The types we support are:
+  * `build`: changes that affect the build system or external dependencies
+  * `chore`: changes that do not modify source or test files
+  * `ci`: changes to our CI configuration files and scripts
+  * `docs`: documentation additions or updates
+  * `feat`: new features or capabilities added to the agent
+  * `fix`: bugfixes or corrections to existing functionality
+  * `perf`: performance improvements
+  * `refactor`: changes that do not add new feature or fix bugs, but improve code structure or readability
+  * `revert`: revert a previous commit
+  * `security`: changes related to the security of the agent, including the updating of dependencies due to CVE
+  * `style` - changes that do not affect the meaning of the code (e.g. formatting, white-space, etc.)
+  * `test` - adding new tests or modifying existing tests
+* Use the imperative, present tense (e.g. "add feature" instead of "added feature")
+* Optionally, include a scope in parantheses after the type to indicate which part of the repository is affected (e.g. `feat(instrumentation): add support for Prisma Client`)
+
+Please note that we only require commits to the main branch to adhere to our Conventional Commit standards. Commits to other branches or in forks may follow a different convention as long as they are clear and descriptive. We use the [Squash and Merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) method when merging Pull Requests into the main branch. This means that all commits in a Pull Request will be combined into a single commit before merging. The commit message for this single commit must be Conventional, and will be used in our automated release note generation process.
```

Comment:
after reading that how does this work? the last commit will be the initial line and then all subsequent commits will be the additional information which do not have to be conventional commit style?

Reply:
Yeah, I think I just caused more confusion explaining how squash and merge works. For this, when you squash and merge, it'll use the PR title and description to fill out the commit message, it won't use the original commits. What do you think about this instead:

```
Please note that we use the Squash and Merge method when merging uppl requests into the main branch. We do not use the original commit messages from each individual commit. Instead, we use the Pull Request title and description as the commit message for the squashed commit, and require that the Pull Request title adheres to our Conventional Commit standards.
```

Reply:
oh wait so the description is what's being used at the body of the conventional commit?  My previous comments about keeping links/description is going to be problematic.  If so, we need to tweak this. we need a place to provide context that doesn't go into the changelog but more for the reviewers and posterity

Reply:
<img width="1728" alt="Screenshot 2023-05-11 at 12 09 06 PM" src="https://github.com/newrelic/node-newrelic/assets/3333180/8b9f9bbe-8d06-4964-887a-e46df0a60d68">

These are the options for the default squash and merge commit message, we could just default to the PR title, and then the body will be blank, leaving the person merging it the ability to write whatever necessary information they want in the release notes there, and then the PR description is just the PR description, what're your thoughts about doing it that way?

Reply:
For future us: Bob and I talked on Zoom and switched to using `Default to pull request title`. The squash commit body will be empty by default, and will be the place to put extended information for the release notes on merge, the PR body will be for reviewers.